docs/design/queue-resource-reservation-design.md

```diff
@@ -84,27 +81,47 @@ List of locked nodes.
 Total idle resource of locked nodes. You can judge whether reservation has met demand by watching this field.
 
 #### Algorithm
-##### Node Selection
-* Big Nodes
 
-Sort all nodes according total resource amount of one or some dimensions. Select the least nodes which rank first and 
-satisfy requirement. It's the most efficient way for preemption reservation.  
+Sort all nodes combination which satisfy the queue request resource in all dimensions.
 
-* Idle First
+and these situations will effect the order of the nodes  combination.
 
-Sort all nodes according idle resource of one or some dimensions. Select the least nodes which rank first as locked nodes.
-It's suitable for non-preemption. It's should to be noted that the locked nodes are often optimal solution but suboptimal
-solution sometimes. 
+* sum(nodes's resource)
 
-##### Lock Strategy
-* Disposable Lock
+the sum of the combination nodes's resource is lower,the priority is higher. and the weight is 0.4
+
+* queue already used
+
+the more resources are used on the nodes that belong to the queue,the priority is higher. and the weight is 0.35
+
+* sum(nodes's number)
+
+the nums of the combination nodes is lower,the priority is higher. and the weight is 0.15
+
+* sum(nodes's idle resource)
 
-Disposable Lock to all selected nodes. It's certainly effective for preemption reservation. To non-preemption reservation,
-It' a good practice but not always.
+the more idle resources on the nodes,the priority is higher. and the weight is 0.1
 
-* Multiple Lock
+* Complete formula 
+
+0.4*1/(sum-target)/[(sum-target)+used+n+idle] + 0.35*used/[(sum-target)+used+n+idle] + 0.15*1/n/[(sum-target)+used+n+idle] + 0.1*idle/[(sum-target)+used+n+idle]
```

Comment:
Please add some comments of all parameters and make a breif introduction of the formula